ed25519/signer.go

```diff
@@ -0,0 +1,100 @@
+package ed25519
+
+import (
+	"context"
+	"crypto"
+	"crypto/ed25519"
+	"fmt"
+
+	wrapping "github.com/hashicorp/go-kms-wrapping/v2"
+)
+
+// Signer provides and ed25519 implementation for the wrapping.Signer interface
+type Signer struct {
+	privKey     ed25519.PrivateKey
+	keyId       string
+	keyPurposes []wrapping.KeyPurpose
+	keyType     wrapping.KeyType
+}
+
+var _ wrapping.SigInfoSigner = (*Signer)(nil)
+
+// NewSigner creates a new Signer.   Supported options: WithKeyId,
+// WithKeyPurposes, WithPrivKey
+func NewSigner(ctx context.Context, opt ...wrapping.Option) (*Signer, error) {
+	const op = "crypto.NewSigner"
+	opts, err := getOpts(opt...)
+	if err != nil {
+		return nil, fmt.Errorf("%s: %w", op, err)
+	}
+	if len(opts.WithKeyPurposes) == 0 {
+		opts.WithKeyPurposes = append(opts.WithKeyPurposes, wrapping.KeyPurpose_Sign)
+	}
+	return &Signer{
+		privKey:     opts.WithPrivKey,
+		keyId:       opts.WithKeyId,
+		keyPurposes: opts.WithKeyPurposes,
+		keyType:     wrapping.KeyType_ED25519,
+	}, nil
+}
+
+// SetConfig sets the fields on the Signer
+//
+// Supported options: wrapping.WithKeyId, wrapping.WithKeyPurposes,
+// wrapping.WithConfigMap and the local WithPrivKey
```

Comment:
Could you document that using the local WithPrivKey will override any private key provided in the config map?

Reply:
done!